File: No.701-800/_724. Find Pivot Index/src/Solution.java
import java.util.Arrays;

/**
 * @author Yu-Jing
 * @create 2023/2/8 下午 02:44
 */
public class Solution {
    public int pivotIndex(int[] nums) {
        if (nums.length == 1) return 0;

        for (int i = 0; i < nums.length; i ++){
            int[] arrRight = Arrays.copyOfRange(nums, i + 1, nums.length);

            if (i == 0) {
                if (0 == sumArr(arrRight)) return 0;

            } else {
                int[] arrLeft = Arrays.copyOfRange(nums, 0, i);
                if (sumArr(arrLeft) == sumArr(arrRight)) return i;
            }
        }
        return -1;
    }

    public int sumArr(int[] nums){
        int sum = 0;
        for (int n : nums){
            sum += n;
        }
        return sum;
    }
}

File: No.701-800/_724. Find Pivot Index/kotlin/Solution.kt
/**
@author Yu-Jing
@create 2023-10-01-下午 06:05
 */
class Solution724(){
    fun pivotIndex(nums: IntArray): Int {
        val sum = nums.sum()
        var leftSum = 0
        for (i in nums.indices) {
            if (leftSum == sum - leftSum - nums[i]) {
                return i
            }
            leftSum += nums[i]
        }


        return -1
    }
}